src/main/java/com/microsoft/graph/http/GraphServiceException.java

```diff
@@ -377,7 +381,7 @@ public GraphError getServiceError() {
         }
 
         final String responseMessage = response.message();
-        String rawOutput = "{}";
+        String rawOutput;
 
         InputStream is = response.body().byteStream();
```

Comment:
couldn't we make a copy of the stream before we close it? This way we'd avoid having to get the bytes from the stream later on

Reply:
We do actually make a copy (of the rawOutput).

Reply:
allow me to rephrase, couldn't we use a copy of the input stream instead of doing `rawOutput.getBytes(UTF_8)` down below?

Reply:
We can get all of the bytes out of the stream, and then store those in a byte array. And then we can feed the byte array into serializer.deserializeObject() and get the error. But we still need the stringified response, so we can log the failure to deserialize the error. I might be misunderstanding, but it's almost the same, just the other way around (get the string, and the bytes out of it; or get the bytes, and then create the string)

Reply:
almost the same, but this way will save us from having to get the bytes from the rawOutput (which I believe is o(n) ?)

Reply:
Yeah, you're right there. The ByteArrayInputStream can use the byte array directly, and we can basically reuse the same byte array for the string. Thanks for the suggestion!

Reply:
Done.

Reply:
thanks, resolving this thread